clients/redigo.go

```diff
@@ -115,7 +115,23 @@ func (r *Redigo) JSONType(key, path string) (res interface{}, err error) {
 	if err != nil {
 		return nil, err
 	}
-	return r.Conn.Do(name, args...)
+
+	res, command_err := r.Conn.Do(name, args...)
+
+	if err != nil {
+		return nil, command_err
+	}
+	switch v := res.(type) {
+	case string:
+		return v, nil
+	case []uint8:
+		return string(v), nil
+	case nil:
+		return
+	default:
+		err := fmt.Errorf("type returned not expected %T", v)
+		return nil, err
+	}
```

Comment:
Yes, I agree with this. Just a few suggestions, it would be better if we directly return the value in case of default and we should also handle the `[]byte` type.

And I don't think `nil` check should be required, is there something I am missing here?

Reply:
There is some tests regarding nil inputs and they would break if not handled properly

https://github.com/nitishm/go-rejson/blob/master/rejson_test.go#L763-L770

Reply:
About the byte agree ill make the change, thought []uint8 would cover

Reply:
@Shivam010 👀  
![image](https://user-images.githubusercontent.com/13908455/149622647-e7d61564-fea8-4f43-805f-b19667dc8126.png)

Reply:
Maybe I leave just the []byte, more straight forward 😅

Reply:
Yes, change it to []byte, that would be more clear